src/reformat.c

```diff
@@ -151,7 +151,7 @@ static avifBool avifPrepareReformatState(const avifImage * image, const avifRGBI
     }
 
     state->toRGBAlphaMode = AVIF_ALPHA_MULTIPLY_MODE_NO_OP;
-    if (image->alphaPlane) {
+    if (!avifImageIsOpaque(image)) {
```

Comment:
This change needs to be reviewed carefully. I suggest excluding this change from this pull request so that we can merge the rest of the pull request immediately.

For example, this function (`avifPrepareReformatState()`) is also called by `avifImageRGBToYUV()`. In the case of `avifImageRGBToYUV()`, the `image` parameteter is the output, so we should not read its alpha pixels.

I am wondering if you made this change to prevent us from going to line 159 unnecessarily when doing `avifImageYUVToRGB()`?

Reply:
> This change needs to be reviewed carefully. I suggest excluding this change from this pull request so that we can merge the rest of the pull request immediately.

Done.

> For example, this function (`avifPrepareReformatState()`) is also called by `avifImageRGBToYUV()`. In the case of `avifImageRGBToYUV()`, the `image` parameteter is the output, so we should not read its alpha pixels.

I missed that. It works because only `state->toRGBAlphaMode` is set and ignored by `avifImageRGBToYUV()`, but I agree alpha samples should not be read.

> I am wondering if you made this change to prevent us from going to line 159 unnecessarily when doing `avifImageYUVToRGB()`?

Yes. Opaque alpha and missing alpha should have the same output, no matter the `avifAlphaMultiplyMode`.
Unfortunately `toRGBAlphaMode != NO_OP` results in different YUV-to-RGB conversion algorithms being used (slow built-in for opaque alpha and libyuv for missing alpha). See #1246.